--- a/versioneer.py
+++ b/versioneer.py
@@ -339,7 +339,7 @@ def get_config_from_root(root):
     # This might raise OSError (if setup.cfg is missing), or
     # configparser.NoSectionError (if it lacks a [versioneer] section), or
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
-    parser = configparser.SafeConfigParser()
+    parser = configparser.RawConfigParser()  # SafeConfigParser removed in Python 3.12
     parser.read(os.path.join(root, "setup.cfg"))
     parser.get("versioneer", "VCS")  # raise error if missing
